xarray/core/coordinates.py

```diff
@@ -397,6 +415,25 @@ def _ipython_key_completions_(self):
         return self._data._ipython_key_completions_()
 
 
+def drop_coords(coords_to_drop, variables, indexes):
+    """Drop index variables associated with variables in coords_to_drop."""
+    for key in set(coords_to_drop) & set(indexes):
+        maybe_midx = indexes[key]
+        idx_coord_names = set(maybe_midx.index.names + [maybe_midx.dim])
+        if isinstance(maybe_midx, PandasMultiIndex):
+            warnings.warn(
+                f"Updating MultiIndexed coordinate {key!r} would corrupt indices for "
+                f"other variables: {list(maybe_midx.index.names)!r}. "
+                f"This will raise an error in the future. Use `.drop_vars({idx_coord_names!r})` before "
+                "assigning new coordinate values.",
+                DeprecationWarning,
+                stacklevel=4,
```

Comment:
Should be 4 for `assign_coords` and 5 for `Dataset.assign` but perhaps this is OK.